### PR TITLE
Shippo Connector

### PR DESCRIPTION
**Pull request by ctvivek**
Closes https://github.com/ethyca/fides/issues/1877

### Code Changes
NEW SaaS Connector Shippo for access

### Steps to Confirm

* [x] Add/configure the Shippo connector in UI
* [ ] Verify Shippo config and icon UI looks good 
* [x] Add customer data using Shippo UI (no api to add sales) 
* [x] Create an Access DSR with the new Shippo customer email 
* [x] Verify email sent by fides with customer data is correct

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [ ] Update `CHANGELOG.md`

### SaaS Connector Checklist 
* [x] [Spec doc link](https://ethyca.atlassian.net/wiki/spaces/EN/pages/2595029001/Shippo+Connector+Specification)
* [x] Vault Creds
* [x] Postman Collection
* [x] SaaS Config And Dataset 
* [x] Access Tests Passing 
* [ ] Icon

### CODE REVIEW

**Comment by ctvivek** [comment]
Screenshot from 2023-03-02 21-34-17

**Comment by ctvivek** [comment]
Screenshot from 2023-03-02 21-34-48

**Comment by ctvivek** [comment]
Screenshot from 2023-03-02 21-36-08

File: data/saas/config/shippo_config.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+saas_config:
+  fides_key: <instance_fides_key>
+  name: Shippo SaaS Config
```
Comment (galvana) [inline data/saas/config/shippo_config.yml:3]: Can you rename this to `Shippo`? We won't have the saas_connector_registry very soon and this is where we will pull the name to use in the UI?

File: data/saas/dataset/shippo_dataset.yml
Comment (galvana) [inline data/saas/dataset/shippo_dataset.yml:1]: Good work with the data categories!

File: tests/ops/fixtures/saas/shippo_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+def shippo_create_erasure_data(
+    shippo_connection_config: ConnectionConfig, shippo_erasure_identity_email: str
+) -> None:
+
+    sleep(60)
+
+    shippo_secrets = shippo_connection_config.secrets
+    base_url = f"https://{shippo_secrets['domain']}"
+    headers = {
+        "Authorization": f"ShippoToken {jira_secrets['api_key']}",
```
Comment (galvana) [inline tests/ops/fixtures/saas/shippo_fixtures.py:120]: Looks like a copy paste error here, can you rename `jira_secrets` to `shippo_secrets`

File: tests/ops/fixtures/saas/shippo_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    dataset.delete(db=db)
+    ctl_dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def shippo_create_erasure_data(
+    shippo_connection_config: ConnectionConfig, shippo_erasure_identity_email: str
+) -> None:
+
+    sleep(60)
```
Comment (galvana) [inline tests/ops/fixtures/saas/shippo_fixtures.py:115]: What's the purpose of this sleep? We might be able to use the `poll_for_existence` function

File: tests/ops/integration_tests/saas/test_shippo_task.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import random
+import time
+
+import pytest
+import requests
```
Comment (galvana) [inline tests/ops/integration_tests/saas/test_shippo_task.py:5]: Make sure you get rid of the unused imports, the Pylance module in VS Code should help, we can go over the installation during our working sessions

File: tests/ops/integration_tests/saas/test_shippo_task.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+@pytest.mark.integration_shippo
+@pytest.mark.asyncio
+async def test_shippo_access_request_task(
+    db,
+    policy,
+    shippo_connection_config,
+    shippo_dataset_config,
+    shippo_identity_email,
+) -> None:
+    """Full access request based on the shippo SaaS config"""
```
Comment (galvana) [inline tests/ops/integration_tests/saas/test_shippo_task.py:34]: nit: capitalize Shippo when used in a sentence

File: tests/ops/fixtures/saas/shippo_fixtures.py (hunk 3 above)
Comment (Kelsey-Ethyca) [reply tests/ops/fixtures/saas/shippo_fixtures.py]: sleep was removed

**Review by galvana — APPROVED** [review]

**Pull request merged**